### PR TITLE
Support `list` iterable in `For` loop

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
Closes: #926

### CODE REVIEW

File: integration_tests/test_list_01.py [hunk 1]
```diff
@@ -18,4 +18,20 @@ def test_list():
     for i in range(15):
         assert (f[i] - a[i]) == (i + 1.0)
 
-test_list()
+def test_list_02():
+    x: list[f64] = [1.0, 2.0]
+    y: list[f64] = []
```
Comment (czgdp1807) [inline integration_tests/test_list_01.py:23]: Let's make this test a little bit stronger. Doing appends to `x` in a loop after this and then do the actual checking.

File: integration_tests/test_list_01.py [hunk 2]
```diff
@@ -18,4 +18,20 @@ def test_list():
...
+    i: f64
+
+    for i in x:
+        y.append(i)
+
+    j: i32
+    for j in range(len(y)):
```
Comment (czgdp1807) [inline integration_tests/test_list_01.py:30]: `len(y)` works in your branch?

Comment (Thirumalai-Shaktivel) [reply integration_tests/test_list_01.py:30]: Nah, Not yet. We need to wait until the #930 is merged.

**Review by Smit-create — APPROVED** [review]
LGTM, thanks!

File: integration_tests/test_list_01.py [hunk 3]
```diff
@@ -18,4 +18,23 @@ def test_list():
...
         assert (f[i] - a[i]) == (i + 1.0)
 
-test_list()
+def test_list_02():
+    x: list[i32] = [1, 2]
+    y: list[i32] = []
+
+    i: i32
+    for i in range(3, 50):
+        x.append(i)
```
Comment (czgdp1807) [inline integration_tests/test_list_01.py:27]:
```suggestion
        x.append(i + 29)
```

Comment (Thirumalai-Shaktivel) [reply integration_tests/test_list_01.py]: Done!

**Comment by czgdp1807** [comment]
@Thirumalai-Shaktivel Can you try testing the same feature for list of strings as well? I am hoping that it will work but still.

**Comment by Thirumalai-Shaktivel** [comment]
Yup, We should test string as well.

**Review by czgdp1807 — APPROVED** [review]

**Comment by Thirumalai-Shaktivel** [comment]
Thanks @Smit-create and @czgdp1807 for the review!

**Review by certik — APPROVED** [review]

**Comment by Thirumalai-Shaktivel** [comment]
Thank you!

**Pull request merged**